### PR TITLE
Remove redundant ticker.minds

### PR DESCRIPTION
**Pull request by adacovsk**
Nervere said he would be doing stuff with `ticker.minds` in the future, so it will indeed serve a purpose. **Tested** **it** and everything is indeed redundant because of `mob/living/Login()` containing `mind_initialize()`. All living/mobs call Login() when keys are assigned to mobs. Wish someone about mind_initialize in living/Login earlier... Here I was thinking `ticker.minds += L.mind` was actually important

## Why it's good
- Less clutter and redundant code

### CODE REVIEW

**Comment by jwhitak** [comment]
ticker.minds still exists in the following files:

- code/datums/gamemode/objectives/target/target.dm
- code/datums/statistics/stat_helpers.dm
- code/__HELPERS/game.dm
- code/datums/mind.dm

Disregard, this PR removes *redundant* ticker.minds entries

**Comment by Kromkar** [comment]
Did you test this

**Comment by Eneocho** [comment]
> Did you test this

Bro literally in the first line: "**Tested it** and everything is indeed redundant"

**Comment by d3athrow** [comment]
i don't believe you

File: code/controllers/subsystem/supply_shuttle.dm [hunk 1]
```diff
@@ -482,15 +482,13 @@ var/datum/subsystem/supply_shuttle/SSsupply_shuttle
...
-		M.apply_disguise()
 	for(var/mob/living/M in contents)
-		if(M.key || M.ckey) //only mobs that were never player controlled
+		if(istype(M, /mob/living/simple_animal/hostile/mimic))
+			var/mob/living/simple_animal/hostile/mimic/mimic = M
+			mimic.angry = 0
+			mimic.apply_disguise()
+		if(M.key || M.ckey || M.mind) //only mobs that were never player controlled
 			return TRUE
-		for(var/datum/mind/M2 in ticker.minds) //see above, but for ghosts
```
Comment (nervere) [inline code/controllers/subsystem/supply_shuttle.dm:491]: Why remove this? I don't get why you feel the need to touch this file, anyway, it's not like it contains any redundant `ticker.minds += xyz...` which is what we were discussing.

File: code/controllers/subsystem/supply_shuttle.dm [hunk 2]
```diff
@@ -482,15 +482,13 @@ var/datum/subsystem/supply_shuttle/SSsupply_shuttle
...
-			if(M2.current == M)
-				return TRUE
```
Comment (nervere) [inline code/controllers/subsystem/supply_shuttle.dm:493]: Why remove this? I don't get why you feel the need to touch this file, anyway, it's not like it contains any redundant `ticker.minds += xyz...`, seems unrelated to the goal of this PR.

**Review by DamianX — APPROVED** [review]

File: code/controllers/subsystem/supply_shuttle.dm (hunk 2 above)
Comment (adacovsk) [reply code/controllers/subsystem/supply_shuttle.dm:493]: it's not needed and easily replaced with M.mind, instead of looping through a whole list of minds, you just need to check if it has a mind

**Comment by ShiftyRail** [comment]
Damian approved it, so I guess it's worth merging now.
Fix 
```
DEBUG: Runtime in powernet.dm, line 93: undefined variable /obj/machinery/atmospherics/miner/toxins/var/powernet 

proc name: add connection (/datum/powernet/proc/add_connection)
src: /datum/powernet (/datum/powernet)
call stack:
/datum/powernet (/datum/powernet): add connection(the Plasma Gas Miner (/obj/machinery/atmospherics/miner/toxins))
the Plasma Gas Miner (/obj/machinery/atmospherics/miner/toxins): update powernet()
the Plasma Gas Miner (/obj/machinery/atmospherics/miner/toxins): power change()
Atmospherics (/area/engineering/atmos): power change()
Atmospherics APC (/obj/machinery/power/apc): update()
Atmospherics APC (/obj/machinery/power/apc): process()
Power (/datum/subsystem/power): fire(0)
Power (/datum/subsystem/power): ignite(0)
Master (/datum/controller/master): RunQueue()
Master (/datum/controller/master): Loop()
Master (/datum/controller/master): StartProcessing()
``` 

before.

**Comment by ShiftyRail** [comment]
image

A French presidential candidate, on a TV debate related to some French rules, once said "when it's not broken, don't fix it."
She got 2.30% of the votes. But that doesn't mean she wasn't right. If you want to fix something, show that it's broken.

**Comment by adacovsk** [comment]
This PR removes the shitcode. I thought the list was populated using it, but it's actually just redundant code that isn't used.

**Comment by Kurfursten** [comment]
As far as I can tell there's nothing problematic with this from a code perspective. I think people are suspicious of it because of some past PRs having bugginess.

My only point of confusion is that I don't see you removing the definition of minds within ticker. Nothing is being added to it anymore, right?

**Comment by adacovsk** [comment]
> My only point of confusion is that I don't see you removing the definition of minds within ticker. Nothing is being added to it anymore, right?

All minds are added to `ticker.minds` when `mind_initialize()` is called in `mob/living/Login()`. This accounts for every mob with a mind already. The mind is added to ticker.minds whenever a player takes control of a mob already. I made one small change to kanef's old code too... It's more readable and slightly faster.

**Comment by Kurfursten** [comment]
Okay, makes sense to me
